### ISSUE TITLE
[Bug]: Manifest V3 extensions wrongly shows mandatory permissions as optional

### ISSUE THREAD

**Issue by Juraj-Masiar** [mozilla/addons#15103]
### What happened?

Originally in Manifest V3, all host permissions were meant to be Optional.
This was changed however and now they are granted upon installation.
For example, my own extension Search Result Previews:
https://addons.mozilla.org/en-US/firefox/addon/search_result_previews/
See manifest.json file:
https://robwu.nl/crxviewer/?crx=https%3A%2F%2Faddons.mozilla.org%2Fen-US%2Ffirefox%2Faddon%2Fsearch_result_previews%2F%3Futm_source%3Dhome

As you can see, it requires:
![image](https://github.com/user-attachments/assets/e336c091-1ca7-4e98-a253-250eb0272d63)
and one content script:
![image](https://github.com/user-attachments/assets/1abad15d-1de3-4799-81bc-88deffb43d8d)

But on the AMO, only the content script HOST is listed in the mandatory list:
![image](https://github.com/user-attachments/assets/00141791-9f67-43d1-a68b-3da9fa0c3e02)

### What did you expect to happen?

I see 3 issues here:
1. same domain (in this case `"*://*.fastaddons.com/*"`) should not be listed in both lists.
2. all `host_permissions` domains should be listed in the "This add-on needs to:" list:
```
    "*://*.google.com/*",
    "*://*.duckduckgo.com/*",
    "*://*.fastaddons.com/*"
```
3. the `optional_host_permissions` is completely ignored. Shouldn't these be in the "This add-on may also ask to:" list?

### Is there an existing issue for this?

- [X] I have searched the existing issues

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/AMOENG-1157)

**Comment by willdurand** [comment]
> the `optional_host_permissions` is completely ignored. Shouldn't these be in the "This add-on may also ask to:" list?

I don't think we support this field yeah, since that was added in Fx128. This should be its own issue, though.

**Comment by willdurand** [comment]
> all `host_permissions` domains should be listed in the "This add-on needs to:" list:

I don't think I agree with this, because the host permissions are still optional even if they are now granted at install time. Firefox still lists these permissions in the list of optional permissions too. That's because users can revoke the permissions if they want to (unlike required permissions).

**Comment by Juraj-Masiar** [comment]
Users are most likely interested to know, what pages the addon is getting access to when installed.
At least that has been the behavior till now in MV2.
So if it says now that it _may also ask for permission to a page_, but in fact, it already has access to it the moment I install it, it's just a lie.

To compare with Chrome Store:
![image](https://github.com/user-attachments/assets/36b2e9fe-97b4-46cf-9985-5a2e0a3ff057)

Edge Store will also list all host permissions (in Edge I ask also for Bing, so it has 4 hosts):
![image](https://github.com/user-attachments/assets/d7c86523-191f-4010-a8fd-6df56ccfad9e)

**Comment by eviljeff** [comment]
@topotropic to look into the UX for this

**Comment by topotropic** [comment]
Sorry for the delay here, I just tried this in Nightly and got the following install prompt 
<img width="473" alt="image" src="https://github.com/user-attachments/assets/b97aa558-2edd-4c7d-bbcb-1f340fcf2294" />

And based on the description, it looks like this is what's expected. Please ping me if this isn't the case. Thanks!

**Comment by Juraj-Masiar** [comment]
@topotropic this bug is not about the prompt - which shows the correct info - that the addon can access those 3 domains.
The issue is that the AMO page is showing them as optional:

![image](https://github.com/user-attachments/assets/7351eb52-a588-491f-af1b-612e90f73e59)

Saying "This add-on may also ask to:" is not true, because the addon already has access to those.

**Comment by Juraj-Masiar** [comment]
The problem is still here:

1. list of mandatory hosts doesn't match
2. list of optional is wrong and repeats hosts from mandatory list

<img width="1337" height="537" alt="Image" src="https://github.com/user-attachments/assets/3fc79911-5b8f-4001-9d00-ed41870c3fa7" />
